bgpd/bgp_route.c

```diff
@@ -6669,14 +6666,20 @@ void bgp_aggregate_decrement(struct bgp *bgp, struct prefix *p,
 #define AGGREGATE_AS_SET       1
 #define AGGREGATE_AS_UNSET     0
 
-static const char *bgp_origin2str(uint8_t origin)
+const char *bgp_origin2str(uint8_t origin, bool origin_short)
 {
 	switch (origin) {
 	case BGP_ORIGIN_IGP:
+		if (origin_short)
+			return "i";
 		return "igp";
```

Comment:
json output should stay IGP, EGP and INCOMPLETE else we will cause issues :(

Reply:
you mean uppercased?

Reply:
yes. otuput shoudl stay the same here else scripts/json will break

Reply:
Actually this is strange and mixed lots of different ways.. `egp`, `IGP`, json output has `incomplete` while _should_ be `INCOMPLETE` according to the format (uppercased). How do you recommend proceeding here?

Reply:
@ton31337 the case change is also what is triggering the labn ci failures.I suggested skipping the case change here -- and coming up with a PR that provides consistency across bgpd once this has been discussed and agreed upon in a dev meeting (I'm in favor of doing such a change).

Reply:
any consensus regarding this?

Reply:
IMHO, these should be uppercase... and not mixed. The problem is folks might have written scripts that expect one or the other?

Reply:
That's right, but having lowercased, uppercased and short is kinda strange, IMO. It's hard to fix something without a deprecation period here.